Refactor state extraction to use typed StateRecord and enums

cyiallou opened the pull request:
- Introduced `StateRecord`, a typed NamedTuple, to replace dict-based state records.
- Replaced raw integer state values with proper enum names using `ComponentStateCode` and `ComponentErrorCode`.
- Improved internal structure by splitting `_extract_state_records()` and `_filter_alerts()` into cleaner, testable units.
- Added `_resolve_enum_name()` to map proto values to readable enum names.
- Updated unit tests accordingly for enum-based comparisons and removed deprecated test cases.

This improves downstream use of state records, such as in alert visualisation and reporting.

cyiallou committed:
Refactor state extraction to use typed StateRecord and enums

- Introduced `StateRecord`, a typed NamedTuple, to replace dict-based
  state records.
- Replaced raw integer state values with proper enum names using
  `ComponentStateCode` and `ComponentErrorCode`.
- Improved internal structure by splitting `_extract_state_records()`
  and `_filter_alerts()` into cleaner, testable units.
- Added `_resolve_enum_name()` to map proto values to readable enum names.
- Updated unit tests accordingly for enum-based comparisons and removed
  deprecated test cases.

This improves downstream use of state records, such as in alert
visualisation and reporting.

Signed-off-by: cyiallou - Costas <42914163+cyiallou@users.noreply.github.com>

cyiallou commented on src/frequenz/reporting/_reporting.py:
```diff
@@ -192,30 +225,23 @@ async def fetch_and_extract_state_durations(
         include_bounds=False,
```
check again

flora-hofmann-frequenz reviewed (commented):
LGTM at first glace. The naming likely to change with updates in `client-common` though (also a topic for assets api that I would like to discuss next week). If you need this now it is probably good to move along, but we will have to rename once changes from the 'api-common' side are pulled through.

cyiallou commented:
> If you need this now it is probably good to move along

This PR makes it easier to work with downstream because as it is now it accepts a list of integers as `alert_states` which means that you already need to know the Enum values from the `ComponentStateCode` so that you can pass them in and then you need to convert these values back to the Enum name to improve readability - which is not really convenient.

This code is used in the Alerts notebook on `lib-notebooks`.

cyiallou committed:
Raise minimum frequenz-client-common version to v0.3.3

The codebase requires `frequenz.client.common.enum_proto`, which was
added in v0.3.3. Earlier versions lack this module, causing test failures
in `pytest_min`.

Bumping the minimum version ensures compatibility with
`frequenz-repo-config`’s  min-dependency testing and avoids runtime
import errors.

Signed-off-by: cyiallou - Costas <42914163+cyiallou@users.noreply.github.com>

cyiallou commented:
Closing as this code has been moved to `frequenz-lib-notebooks` with this PR: https://github.com/frequenz-floss/frequenz-lib-notebooks/pull/131

I will open another PR that will remove the parts of the code that have been moved.
